varnishgather

```diff
@@ -428,6 +436,21 @@ done
 
 shift $((OPTIND - 1))
 
+if [ -z "$NAME" ]; then
+    INSTANCE_NAME=$(get_varnishd_instance)
+    [ -n "$INSTANCE_NAME" ] && NAME="-n $INSTANCE_NAME"
+else
+    INSTANCE_NAME="${NAME#-n }"
+fi
+
+if [ -n "$NAME" ] && ! validate_instance; then
+    INSTANCE_NAME=$(get_varnishd_instance)
+    [ -n "$INSTANCE_NAME" ] && NAME="-n $INSTANCE_NAME" || NAME=""
+fi
+
+ID="$(cat /etc/hostname)-$(date +'%Y%m%d-%H%M%S')${INSTANCE_NAME:+-$(n_opt "$INSTANCE_NAME")}"
```

Comment:
isn't that just?
```
ID="$(cat /etc/hostname)-$(date +'%Y%m%d-%H%M%S')$(n_opt "$INSTANCE_NAME")"
```
since `n_opt` will be a noop if `INSTANCE_NAME` is empty?

Reply:
The reason I have it the way it is.  When using just ID="$(cat /etc/hostname)-$(date +'%Y%m%d-%H%M%S')$(n_opt "$INSTANCE_NAME")"
the results from a named instance output look like below.
Please submit the file: varnishgather.varnish-wrk-20241101-184253varnish1.tar.gz

Reply:
what I'm saying is that `$(n_opt "$INSTANCE_NAME")"` is a shorter equivalent to `${INSTANCE_NAME:+-$(n_opt "$INSTANCE_NAME")}"`